pulsar-broker/src/test/java/org/apache/pulsar/broker/transaction/pendingack/PendingAckPersistentTest.java

```diff
@@ -99,6 +104,52 @@ protected void cleanup() {
         super.internalCleanup();
     }
 
+    @Test
+    public void testUnloadSubscriptionWhenFailedInitPendingAck() throws Exception {
+        String topic = NAMESPACE1 + "/testUnloadSubscriptionWhenFailedInitPendingAck";
+        Consumer<byte[]> consumer1 = pulsarClient.newConsumer()
+                .subscriptionName("subName1")
+                .topic(topic)
+                .subscribe();
+        // Fail at transactionPendingAckStoreProvider::checkInitializedBefore.
+        Field transactionPendingAckStoreProviderField = PulsarService.class
+                .getDeclaredField("transactionPendingAckStoreProvider");
+        transactionPendingAckStoreProviderField.setAccessible(true);
+        TransactionPendingAckStoreProvider pendingAckStoreProvider =
+                (TransactionPendingAckStoreProvider) transactionPendingAckStoreProviderField
+                        .get(pulsarServiceList.get(0));
+        TransactionPendingAckStoreProvider mockProvider = mock(pendingAckStoreProvider.getClass());
+        // Test retryable exception.
+        when(mockProvider.checkInitializedBefore(any()))
+                .thenReturn(FutureUtil.failedFuture(new ManagedLedgerException("mock fail")))
+                .thenReturn(CompletableFuture.completedFuture(false));
+        transactionPendingAckStoreProviderField.set(pulsarServiceList.get(0), mockProvider);
+        Awaitility.await().until(() -> {
+            pulsarClient.newConsumer()
+                    .subscriptionName("subName2")
+                    .topic(topic)
+                    .subscribe();
+            return true;
+        });
```

Comment:
How the test can be passed? The retry logic will also call `pendingAckStoreProvider.checkInitializedBefore`; it should consistently fail no matter how many times you retry.

Reply:
Only the first attempt will throw a retryable exception.
```
  // Test retryable exception.
  when(mockProvider.checkInitializedBefore(any()))
          .thenReturn(FutureUtil.failedFuture(new ManagedLedgerException("mock fail")))
          .thenReturn(CompletableFuture.completedFuture(false));
```

Reply:
Only the first attempt will throw a retryable exception.
```
  // Test retryable exception.
  when(mockProvider.checkInitializedBefore(any()))
          .thenReturn(FutureUtil.failedFuture(new ManagedLedgerException("mock fail")))
          .thenReturn(CompletableFuture.completedFuture(false));
```

Reply:
Only the first attempt will throw a retryable exception.
```
  // Test retryable exception.
  when(mockProvider.checkInitializedBefore(any()))
          .thenReturn(FutureUtil.failedFuture(new ManagedLedgerException("mock fail")))
          .thenReturn(CompletableFuture.completedFuture(false));
```

Reply:
Only the first attempt will throw a retryable exception.
```
  // Test retryable exception.
  when(mockProvider.checkInitializedBefore(any()))
          .thenReturn(FutureUtil.failedFuture(new ManagedLedgerException("mock fail")))
          .thenReturn(CompletableFuture.completedFuture(false));
```

Reply:
Ah, good to know.